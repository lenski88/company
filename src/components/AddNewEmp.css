
.AddNewEmp {
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  margin-top: 4vh;
  background-color: #0301a8;
  border: 0.1em solid;
  z-index: 1000;
}

.AddNewEmp .label {
    margin-left: 1vw;
}

.AddNewEmp .inputText {
  margin: 1vh;
}
