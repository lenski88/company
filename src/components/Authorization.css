.Authorization {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  border: 0.2em solid  #fbfc55;
  padding: 3em;
  margin-top:25vh
}

.Authorization .inputText {
    border-color: #fbfc55;
    background-color: #fbfc55 ;
    color:#0301a8;
    font-family:  'Press Start 2P', cursive;
    padding: 0.3em;
}
