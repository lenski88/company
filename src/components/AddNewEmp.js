import React from "react";
import PropTypes from "prop-types";

import "./AddNewEmp.css";

class AddNewEmp extends React.PureComponent {
  static propTypes = {
    employes: PropTypes.array.isRequired,
    cbExitAdd: PropTypes.func,
    cbAddEmp: PropTypes.func,
  };

  state = {
    newLogin: "",
    newLevel: 1,
    newFIO: "",
    newEmail: "",
    newPhone: "",
    newDepartment: "Руководство",
    newPosition: "",

    errLogin: "",
    errFIO: "",
    errEmail: "",
    errPhone: "",
    errPosition: "",

    lastId: this.props.employes[this.props.employes.length-1].id
  };

  exitAdd = () => {
    this.props.cbExitAdd();
  };

  setNewLogin = (eo) => {
    this.setState({ newLogin: eo.target.value.toLowerCase().trim() });

    // есть ли такой логин?
    let isHaveLogin = this.props.employes.some((i) => {
      return i.login === eo.target.value;
    });

    if (eo.target.value.length > 7) {
      this.setState({
        errLogin: "Не более 7 символов",
      });
    } else if (isHaveLogin) {
      this.setState({
        errLogin: "Логин уже занят",
      });
    } else {
      this.setState({
        errLogin: "",
      });
    }
  };
  setNewLevel = (eo) => {
    this.setState({ newLevel: Number(eo.target.value) });
  };
  setNewFIO = (eo) => {
    this.setState({ newFIO: eo.target.value });
  };
  setNewEmail = (eo) => {
    this.setState({ newEmail: eo.target.value.trim() });
  };
  setNewPhone = (eo) => {
    this.setState({ newPhone: eo.target.value.trim() });
    let regExp = /^\d+$/;
    if (eo.target.value.length !== 9 || !regExp.test(this.state.newPhone)) {
      this.setState({
        errPhone: "Содержит 9 цифр",
      });
    } else {
      this.setState({
        errPhone: "",
      });
    }
  };
  setNewDepartment = (eo) => {
    this.setState({ newDepartment: eo.target.value });
  };
  setNewPosition = (eo) => {
    this.setState({ newPosition: eo.target.value });
    let regExp = /^[А-ЯЁ].*/;
    if (!regExp.test(eo.target.value)) {
      this.setState({
        errPosition: "Первая буква заглавная; только кириллица",
      });
    } else {
      this.setState({
        errPosition: "",
      });
    }
  };

  validation = (eo) => {
    let name = eo.target.name;

    if (name === "login") {
      if (!/^[a-zA-Z0-9]+$/.test(this.state.newLogin)) {
        this.setState({
          errLogin: "Не может быть пустым; только латинские символы",
        });
      } else {
        this.setState({
          errLogin: "",
        });
      }
    }
    if (name === "FIO") {
      let regExp =
        /^[А-ЯЁ][а-яё]*([-][А-ЯЁ][а-яё]*)?\s[А-ЯЁ][а-яё]*\s[А-ЯЁ][а-яё]*$/;
      if (!regExp.test(this.state.newFIO)) {
        this.setState({
          errFIO: "Введите ФИО полностью",
        });
      } else {
        this.setState({
          errFIO: "",
        });
      }
    }
    if (name === "email") {
      let regExp = /\S+@\S+\.\S+/;
      if (!regExp.test(this.state.newEmail)) {
        this.setState({
          errEmail: "Формат ввода: login@example.com",
        });
      } else {
        this.setState({
          errEmail: "",
        });
      }
    }
  };

  addNewEmp = () => {
    this.props.cbAddEmp({
      id: this.state.lastId + 1,
      login: this.state.newLogin,
      status: 0,
      level: this.state.newLevel,
      name: this.state.newFIO,
      email: this.state.newEmail,
      phone: this.state.newPhone,
      department: this.state.newDepartment,
      position: this.state.newPosition,
      task: [],
    });
    this.props.cbExitAdd();
  };

  render() {
    return (
      <div className="AddNewEmp">
        <h2 style={{ marginLeft: "1vw" }}>Добавление нового сотрудника </h2>
        <br />
        <label className="label">
          Логин:
          <br />
          <input
            className="inputText"
            type="text"
            value={this.state.newLogin}
            name="login"
            placeholder='login'
            onChange={this.setNewLogin}
            onBlur={this.validation}
          ></input>
          <span style={{ color: "black" }}>{this.state.errLogin}</span>
        </label>
        <label className="label">
          Уровень доступа:
          <br />
          <select
            className="inputText"
            name="level"
            onChange={this.setNewLevel}
            onBlur={this.validation}
          >
            <option value={1}>Уровень доступа 1</option>
            <option value={2}>Уровень доступа 2</option>
            <option value={3}>Уровень доступа 3</option>
          </select>
        </label>
        <label className="label">
          ФИО(полностью):
          <br />
          <input
            className="inputText"
            type="text"
            name="FIO"
            placeholder='Иванов Иван Иванович'
            value={this.state.newFIO}
            onChange={this.setNewFIO}
            onBlur={this.validation}
          ></input>
          <span style={{ color: "black" }}>{this.state.errFIO}</span>
        </label>
        <label className="label">
          Почта:
          <br />
          <input
            className="inputText"
            type="text"
            name="email"
            placeholder='login@example.com'
            value={this.state.newEmail}
            onChange={this.setNewEmail}
            onBlur={this.validation}
          ></input>
          <span style={{ color: "black" }}>{this.state.errEmail}</span>
        </label>
        <label className="label">
          Телефон:
          <br />
          <input
            className="inputText"
            type="text"
            name="phone"
            placeholder='333333333'
            value={this.state.newPhone}
            onChange={this.setNewPhone}
            onBlur={this.validation}
          ></input>
          <span style={{ color: "black" }}>{this.state.errPhone}</span>
        </label>
        <label className="label">
          Отдел:
          <br />
          <select
            className="inputText"
            name="department"
            onChange={this.setNewDepartment}
            onBlur={this.validation}
          >
            <option value="Руководство">Руководство</option>
            <option value="Отдел закупок">Отдел закупок</option>
            <option value="Отдел продаж">Отдел продаж</option>
            <option value="Транспортный отдел">Транспортный отдел</option>
            <option value="Бухгалтерия">Бухгалтерия</option>
            <option value="Отдел кадров">Отдел кадров</option>
            <option value="Складское хозяйство">Складское хозяйство</option>
          </select>
        </label>
        <label className="label">
          Должность:
          <br />
          <input
            className="inputText"
            type="text"
            name="position"
            placeholder='Сотрудник отдела'
            value={this.state.newPosition}
            onChange={this.setNewPosition}
            onBlur={this.validation}
          ></input>
          <span style={{ color: "black" }}>{this.state.errPosition}</span>
        </label>
        <br />
        <div>
        {Boolean(this.state.newLogin &&
              this.state.newFIO &&
              this.state.newEmail &&
              this.state.newPhone &&
              this.state.newPosition) &&
              Boolean(!this.state.errLogin &&
              !this.state.errFIO &&
              !this.state.errEmail &&
              !this.state.errPhone &&
              !this.state.errPosition) &&
          <input
            className="button"
            type="button"
            value="[Сохранить]"
            onPointerDown={this.addNewEmp}
          ></input>}
          <input
            className="button"
            type="button"
            value="[Выйти без сохранения]"
            onPointerDown={this.exitAdd}
          ></input>
        </div>
      </div>
    );
  }
}

export default AddNewEmp;
