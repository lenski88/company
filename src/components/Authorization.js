import React from "react";
import PropTypes from "prop-types";

import "./Authorization.css";

class Authorization extends React.PureComponent {
  static propTypes = {
    employes: PropTypes.array.isRequired /* массив сотрудников */,
    login:
      PropTypes.bool.isRequired /* прошла ли авторизация; по умолчанию false */,
    cbGetUser: PropTypes.func,
  };

  state = {
    login: this.props.login,
    nameUser: "",
    errName: "",
    button: false,
  };

  nameUser = (eo) => {
    this.setState({
      nameUser: eo.target.value,
    });

    let arrayNames = this.props.employes;
    let name = eo.target.value;
    let isLogin = arrayNames.some((i) => {
      return name === i.login;
    });

    if (!isLogin) {
      this.setState({
        errName: "Нет такого сотрудника",
      });
    } else {
      this.setState({
        errName: "",
      });
    }
  };

  getUser = () => {
    if (!this.state.errName && !this.state.errPassword) {
      let arrayNames = this.props.employes;
      let name = this.state.nameUser;
      let user = arrayNames.find((i) => {
        return name === i.login;
      });
      this.props.cbGetUser(user);
      this.setState({
        nameUser: "",
        login: this.props.login,
      });
    }
  };

  render() {
    return (
      !this.state.login && (
        <div className={!this.props.login?"Authorization": "NoAuthorization"}>
          <label>Введите логин:</label>
          <br />
          <input
            className="inputText"
            type="text"
            name="nameUser"
            value={this.state.nameUser}
            onChange={this.nameUser}
            autoFocus
          ></input>
          <br />
          <span>{this.state.errName}</span>
          <br />
          <br />
          {this.state.nameUser && (
            <input
              className="button"
              type="button"
              onPointerDown={this.getUser}
              value="[Войти]"
              disabled={this.state.button}
            ></input>
          )}
        </div>
      )
    );
  }
}

export default Authorization;
