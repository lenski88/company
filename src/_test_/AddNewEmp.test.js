"use strict";

import React from "react";
import renderer from "react-test-renderer";
import { employes } from "./Authorization.test";

import AddNewEmp from "../components/AddNewEmp";

test("проверка правильности значений инпутов при открытии", () => {
  let component = renderer.create(<AddNewEmp employes={employes} />);

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let login = component.root.find((el) => el.props.name == "login");
  expect(login.props.value).toBe("");

  let level = component.root.find((el) => el.props.name == "level");
  expect(level.props.value).toBeFalsy();

  let fio = component.root.find((el) => el.props.name == "FIO");
  expect(fio.props.value).toBe("");

  let email = component.root.find((el) => el.props.name == "email");
  expect(email.props.value).toBe("");

  let phone = component.root.find((el) => el.props.name == "phone");
  expect(phone.props.value).toBe("");

  let department = component.root.find((el) => el.props.name == "department");
  expect(department.props.value).toBeFalsy();

  let position = component.root.find((el) => el.props.name == "position");
  expect(position.props.value).toBe("");
});

test("проверка ввода", () => {
  let component = renderer.create(<AddNewEmp employes={employes} />);

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let login = component.root.find((el) => el.props.name == "login");
  login.props.onChange({ target: { value: "asdf" } });

  let level = component.root.find((el) => el.props.name == "level");
  level.props.onChange({ target: { value: 2 } });

  let fio = component.root.find((el) => el.props.name == "FIO");
  fio.props.onChange({ target: { value: "Иванов Иван Иванович" } });

  let email = component.root.find((el) => el.props.name == "email");
  email.props.onChange({ target: { value: "ivanov@example.com" } });

  let phone = component.root.find((el) => el.props.name == "phone");
  phone.props.onChange({ target: { value: "295554422" } });

  let department = component.root.find((el) => el.props.name == "department");
  department.props.onChange({ target: { value: "Отдел продаж" } });

  let position = component.root.find((el) => el.props.name == "position");
  position.props.onChange({ target: { value: "Продавец" } });

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

});

test("проверка некорректного ввода", () => {
  let component = renderer.create(<AddNewEmp employes={employes} />);

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let login = component.root.find((el) => el.props.name == "login");
  login.props.onChange({ target: { value: "asdfsdfasdfasdfsd" } });

  let level = component.root.find((el) => el.props.name == "level");
  level.props.onChange({ target: { value: 2 } });

  let fio = component.root.find((el) => el.props.name == "FIO");
  fio.props.onChange({ target: { value: "ИвановИванИванович" } });

  let email = component.root.find((el) => el.props.name == "email");
  email.props.onChange({ target: { value: "ivanovexample.com" } });

  let phone = component.root.find((el) => el.props.name == "phone");
  phone.props.onChange({ target: { value: "2955544uu22" } });

  let department = component.root.find((el) => el.props.name == "department");
  department.props.onChange({ target: { value: "Отдел продаж" } });

  let position = component.root.find((el) => el.props.name == "position");
  position.props.onChange({ target: { value: "продавец" } });

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

}); 

