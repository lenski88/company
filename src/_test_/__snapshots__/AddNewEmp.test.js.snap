// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка Authorization 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка Authorization 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="sdfasdf"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="saveliev47"
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка ввода 1`] = `
<div
  className="AddNewEmp"
>
  <h2
    style={
      Object {
        "marginLeft": "1vw",
      }
    }
  >
    Добавление нового сотрудника 
  </h2>
  <br />
  <label
    className="label"
  >
    Логин:
    <br />
    <input
      className="inputText"
      name="login"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      className="inputText"
      name="level"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    ФИО(полностью):
    <br />
    <input
      className="inputText"
      name="FIO"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Иванов Иван Иванович"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login@example.com"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="333333333"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Сотрудник отдела"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <br />
  <div>
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Выйти без сохранения]"
    />
  </div>
</div>
`;

exports[`проверка ввода 2`] = `
<div
  className="AddNewEmp"
>
  <h2
    style={
      Object {
        "marginLeft": "1vw",
      }
    }
  >
    Добавление нового сотрудника 
  </h2>
  <br />
  <label
    className="label"
  >
    Логин:
    <br />
    <input
      className="inputText"
      name="login"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login"
      type="text"
      value="asdf"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      className="inputText"
      name="level"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    ФИО(полностью):
    <br />
    <input
      className="inputText"
      name="FIO"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Иванов Иван Иванович"
      type="text"
      value="Иванов Иван Иванович"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login@example.com"
      type="text"
      value="ivanov@example.com"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="333333333"
      type="text"
      value="295554422"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Сотрудник отдела"
      type="text"
      value="Продавец"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <br />
  <div>
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Сохранить]"
    />
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Выйти без сохранения]"
    />
  </div>
</div>
`;

exports[`проверка нажатия на кнопку войти 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка нажатия на кнопку войти 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка некорректного ввода 1`] = `
<div
  className="AddNewEmp"
>
  <h2
    style={
      Object {
        "marginLeft": "1vw",
      }
    }
  >
    Добавление нового сотрудника 
  </h2>
  <br />
  <label
    className="label"
  >
    Логин:
    <br />
    <input
      className="inputText"
      name="login"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      className="inputText"
      name="level"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    ФИО(полностью):
    <br />
    <input
      className="inputText"
      name="FIO"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Иванов Иван Иванович"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login@example.com"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="333333333"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Сотрудник отдела"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <br />
  <div>
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Выйти без сохранения]"
    />
  </div>
</div>
`;

exports[`проверка некорректного ввода 2`] = `
<div
  className="AddNewEmp"
>
  <h2
    style={
      Object {
        "marginLeft": "1vw",
      }
    }
  >
    Добавление нового сотрудника 
  </h2>
  <br />
  <label
    className="label"
  >
    Логин:
    <br />
    <input
      className="inputText"
      name="login"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login"
      type="text"
      value="asdfsdfasdfasdfsd"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      Не более 7 символов
    </span>
  </label>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      className="inputText"
      name="level"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    ФИО(полностью):
    <br />
    <input
      className="inputText"
      name="FIO"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Иванов Иван Иванович"
      type="text"
      value="ИвановИванИванович"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login@example.com"
      type="text"
      value="ivanovexample.com"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="333333333"
      type="text"
      value="2955544uu22"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      Содержит 9 цифр
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Сотрудник отдела"
      type="text"
      value="продавец"
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      Первая буква заглавная; только кириллица
    </span>
  </label>
  <br />
  <div>
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Выйти без сохранения]"
    />
  </div>
</div>
`;

exports[`проверка правильности значений инпутов при открытии 1`] = `
<div
  className="AddNewEmp"
>
  <h2
    style={
      Object {
        "marginLeft": "1vw",
      }
    }
  >
    Добавление нового сотрудника 
  </h2>
  <br />
  <label
    className="label"
  >
    Логин:
    <br />
    <input
      className="inputText"
      name="login"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Уровень доступа:
    <br />
    <select
      className="inputText"
      name="level"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value={1}
      >
        Уровень доступа 1
      </option>
      <option
        value={2}
      >
        Уровень доступа 2
      </option>
      <option
        value={3}
      >
        Уровень доступа 3
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    ФИО(полностью):
    <br />
    <input
      className="inputText"
      name="FIO"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Иванов Иван Иванович"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Почта:
    <br />
    <input
      className="inputText"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="login@example.com"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Телефон:
    <br />
    <input
      className="inputText"
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="333333333"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <label
    className="label"
  >
    Отдел:
    <br />
    <select
      className="inputText"
      name="department"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value="Руководство"
      >
        Руководство
      </option>
      <option
        value="Отдел закупок"
      >
        Отдел закупок
      </option>
      <option
        value="Отдел продаж"
      >
        Отдел продаж
      </option>
      <option
        value="Транспортный отдел"
      >
        Транспортный отдел
      </option>
      <option
        value="Бухгалтерия"
      >
        Бухгалтерия
      </option>
      <option
        value="Отдел кадров"
      >
        Отдел кадров
      </option>
      <option
        value="Складское хозяйство"
      >
        Складское хозяйство
      </option>
    </select>
  </label>
  <label
    className="label"
  >
    Должность:
    <br />
    <input
      className="inputText"
      name="position"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Сотрудник отдела"
      type="text"
      value=""
    />
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      
    </span>
  </label>
  <br />
  <div>
    <input
      className="button"
      onPointerDown={[Function]}
      type="button"
      value="[Выйти без сохранения]"
    />
  </div>
</div>
`;

exports[`проверка рендера кнопки при верном логине 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка рендера кнопки при верном логине 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;
